pkg/plugins/manager/loader/loader.go

```diff
@@ -72,17 +82,52 @@ func (l *Loader) Load(ctx context.Context, src plugins.PluginSource) ([]*plugins
 		}
 		bootstrappedPlugins = append(bootstrappedPlugins, bootstrappedPlugin...)
 	}
+	l.log.Debug("Bootstrapped", "class", src.PluginClass(ctx), "duration", time.Since(st))
 
+	st = time.Now()
 	validatedPlugins := []*plugins.Plugin{}
+	type validateResult struct {
+		bootstrappedPlugin *plugins.Plugin
+		err                error
+	}
+	validateResults := make(chan validateResult, len(bootstrappedPlugins))
+
+	// If the PluginsCDNSyncLoaderEnabled feature is enabled, validate plugins in parallel.
+	// Otherwise, validate plugins sequentially.
+	var limitSize int
+	if l.cfg.Features.PluginsCDNSyncLoaderEnabled && src.PluginClass(ctx) == plugins.ClassCDN {
+		limitSize = len(bootstrappedPlugins)
+	} else {
+		limitSize = 1
+	}
+	limit := make(chan struct{}, limitSize)
```

Comment:
There are a couple of places in this PR where I used this technique to ensure that code runs sequentially if the feature toggle is off. This should save us from race conditions, just in case. Ideally, we should be able to make those bits parallel for all classes (not just CDN), but I wanted to be safe for now

Reply:
Would there be any error if `limitSize = 1` and `bootstrappedPlugins=0`? `validateWg.Wait()` will directly end and close the channel?

Reply:
It should be fine: https://goplay.tools/snippet/_ysOeAzHmTl

However, the `sync.WaitGroup` wasn't actually required, so I have removed it entirely :)